### PR TITLE
Update changelog workflow

### PR DESCRIPTION
**Pull request by epassaro**
## Description
- [x] Add workflow for updating changelog

## Motivation and Context

Before #1326 the `CHANGELOG.md` file was updated only in the `gh-pages` branch and not pushed to `master` via the documentation pipeline. I created a GitHub workflow to automatically update and push the changelog 10 minutes before a new TARDIS release is made. Also, I added a `cron` job to the documentation build (10 minutes after the release).

In the near future, we want to release TARDIS less often.

## How Has This Been Tested?

The first steps of the pipeline were tested (see the Actions tab). The push steps were not tested, we should merge and wait for the `cron` trigger.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: .github/workflows/update-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Install github-changes
+        run: sudo npm install -g github-changes
+
+      - name: Update changelog
+        run: github-changes -o tardis-sn -r tardis --only-pulls --use-commit-body -f CHANGELOG.md -k ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Commit files
+        run: |
+          git config --local user.email "wkerzendorf@gmail.com"
```
Comment (wkerzendorf) [inline .github/workflows/update-changelog.yml:25]: not sure if this is a good idea. It should be the bot.

**Comment by epassaro** [comment]
Ok. I never used the bot. Do you think using the bot mail account should be
enough? Also we should remove ban that email from the author list notebook
(this is a reminder for myself).

**Comment by epassaro** [comment]
Now the workflow commits with the bot account. Sorry for the delay @wkerzendorf

**Review by andrewfullard — APPROVED** [review]

**Comment by epassaro** [comment]
Is ok to merge this? @andrewfullard

**Pull request merged**